### DISCUSSION TITLE
Efficiency of tar_make() with {crew} vs tar_make_clustermq()

### DISCUSSION THREAD

**Discussion by wlandau-lilly** [ropensci/targets#1079] [General]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

This `crew` test with lots of small targets:

https://github.com/ropensci/targets/blob/7463ece43a40ee088de8da67f0cfa406fb4aa189/tests/hpc/test-crew_local.R#L85-L121

is slower than the equivalent `clustermq` test:

https://github.com/ropensci/targets/blob/7463ece43a40ee088de8da67f0cfa406fb4aa189/tests/hpc/test-clustermq_local.R#L150-L181

[`proffer`](https://r-prof.github.io/proffer) flame graphs do not show an overt bottleneck, so it's hard to say what might be slowing things down. It might have to do with my solution to https://github.com/ropensci/targets/discussions/1074. Reverting it seems to shave a couple seconds off.

On the other hand, maybe the above test is not truly representative of the work that takes place in real projects. In a proprietary internal simulation study, `tar_make_clustermq()` took 2.007 minutes, whereas `tar_make()` with the equivalent number of persistent `crew` workers took about 2.661 minutes. From previous profiling studies and from looking at what is in the pipeline, I believe I can explain this difference in terms of serialization. `clustermq` supports "common data", whereas `crew` and `mirai` do not (c.f. https://github.com/wlandau/crew/issues/33).

**Comment by wlandau** [comment]
Ultimately, targets that run quickly are much better off with `deployment = "main"`.

**Comment by wlandau** [comment]
So after further profiling studies, I think the limitation is in `crew` itself rather than the integration of `crew` in `targets`. I will post a separate discussion in the `crew` repo.
